operator/controllers/common/otel_utils.go

```diff
@@ -0,0 +1,285 @@
+	if err != nil {
+		logger.Error(err, "unable to start OTel")
+	}
+
+	appDeploymentDurationGauge, err := meter.AsyncFloat64().Gauge("keptn.app.deploymentduration", instrument.WithDescription("a gauge of the duration of deployments"))
+	if err != nil {
+		logger.Error(err, "unable to start OTel")
+	}
+
+	workloadDeploymentIntervalGauge, err := meter.AsyncFloat64().Gauge("keptn.deployment.deploymentinterval", instrument.WithDescription("a gauge of the interval between deployments"))
+	if err != nil {
+		logger.Error(err, "unable to start OTel")
+	}
+
+	workloadDeploymentDurationGauge, err := meter.AsyncFloat64().Gauge("keptn.deployment.deploymentduration", instrument.WithDescription("a gauge of the duration of deployments"))
+	if err != nil {
+		logger.Error(err, "unable to start OTel")
+	}
+
+	err = meter.RegisterCallback(
+		[]instrument.Asynchronous{
+			deploymentActiveGauge,
+			taskActiveGauge,
+			appActiveGauge,
+			evaluationActiveGauge,
+			appDeploymentIntervalGauge,
+			appDeploymentDurationGauge,
+			workloadDeploymentIntervalGauge,
+			workloadDeploymentDurationGauge,
+		},
+		func(ctx context.Context) {
+			observeActiveInstances(ctx, mgr, deploymentActiveGauge, appActiveGauge, taskActiveGauge, evaluationActiveGauge)
+			observeDeploymentInterval(ctx, mgr, appDeploymentIntervalGauge, workloadDeploymentIntervalGauge)
+			observeDuration(ctx, mgr, appDeploymentDurationGauge, workloadDeploymentDurationGauge)
+		})
+	if err != nil {
+		fmt.Println("Failed to register callback")
+		panic(err)
+	}
+}
+
+func observeDuration(ctx context.Context, mgr client.Client, appDeploymentDurationGauge asyncfloat64.Gauge, workloadDeploymentDurationGauge asyncfloat64.Gauge) {
+
+	err := ObserveDeploymentDuration(ctx, mgr, &lifecyclev1alpha2.KeptnAppVersionList{}, appDeploymentDurationGauge)
+	if err != nil {
+		logger.Error(err, "unable to gather app deployment durations")
+	}
+
+	err = ObserveDeploymentDuration(ctx, mgr, &lifecyclev1alpha2.KeptnWorkloadInstanceList{}, workloadDeploymentDurationGauge)
+	if err != nil {
+		logger.Error(err, "unable to gather workload deployment durations")
+	}
+
+}
+
+func observeDeploymentInterval(ctx context.Context, mgr client.Client, appDeploymentIntervalGauge asyncfloat64.Gauge, workloadDeploymentIntervalGauge asyncfloat64.Gauge) error {
+	err := ObserveDeploymentInterval(ctx, mgr, &lifecyclev1alpha2.KeptnAppVersionList{}, &lifecyclev1alpha2.KeptnWorkloadInstance{}, appDeploymentIntervalGauge)
+	if err != nil {
+		logger.Error(err, "unable to gather app deployment intervals")
+	}
+
+	err = ObserveDeploymentInterval(ctx, mgr, &lifecyclev1alpha2.KeptnWorkloadInstanceList{}, &lifecyclev1alpha2.KeptnWorkloadInstance{}, workloadDeploymentIntervalGauge)
+	if err != nil {
+		logger.Error(err, "unable to gather workload deployment intervals")
+	}
+	return err
```

Comment:
I think here we can also just log the error instead of returning it, as in the other function - this. should also resolve the linting error above

Reply:
done